### DISCUSSION TITLE
How should the OBS dev community use GitHub Discussions?

### DISCUSSION THREAD

**Discussion by WizardCM** [obsproject/obs-studio#3869] [Announcements]
In 2017, we launched the OBS Community Discord server, and with it came the #development channel. One of the goals was to provide a lower barrier of entry for devs to introduce themselves and contribute their thoughts, skills & code. More recently we introduced the Ideas site, and moved Issues from Mantis to GitHub to more actively track tickets.

With GitHub's release of Discussions, we want to lower that barrier even further. If you're already browsing on GitHub, the conversations might as well happen here, right?

So, the question to our contributors, regardless of how much code you've submitted or how many Issues you've opened:

### **How should we, as a community, use GitHub Discussions?**

As it stands, we have the [Discord server](https://obsproject.com/discord), the [Forums](https://obsproject.com/forum/), the [RFC repository](https://github.com/obsproject/rfcs), plus the standard Issues/PRs, not to mention the [Ideas](https://ideas.obsproject.com/) site.

On one hand, this means that there are many places to start a conversation. On the other, this also means conversations can be disjointed. One of the bigger issues is that conversations can be lost into the void, especially due to how active the #development channel is, and how rarely the core devs browse the General Development subforum.

We have a few ideas ourselves, but would love to get a better idea of what our wider community is looking for. Submit your thoughts below! 😃

**Comment by dodgepong** [comment]
What I want to know is **what problems do we currently have that would be solved by GitHub discussions?** The OP mentions lowering barrier to entry and discussions being lost in the void due to our wide variety of places for discussion.

As I see it, GihHub discussions "competes" most directly with the General Development section of the forums. In that sense, the only way that the barrier to entry is lowered is that it doesn't require someone to create another account (in addition to GitHub) to post on those forums. Then again, it's possible there will be higher quality discussion here rather than the forums because posts here send notifications to all watchers, which can be good or bad, depending on the amount of noise. If there's too much noise, then people will disable discussion notifications, in which case the forums are now better because they are at least linked on the main obsproject.com website.

As for conversations getting lost in the void, I do agree that a wide variety of places to post on contributes to that, and making yet another one doesn't help. If we start pushing GitHub discussions, I would advocate for shutting down at least the General Development forum, or redirecting it here.

The biggest question I have is what kind of participation should we expect and invite from non-developers? The Ideas site is effectively a collecting ground for pre-RFC-quality feature proposals largely created by users with no programming ability, and sometimes no grasp of how the program works or what it is capable of. In that sense, I feel like it is useful for keeping developer and non-developer talk separate. I also think the #development channel on Discord is still valuable as a place for real-time developer communication.

In the [RFC submission process](https://github.com/obsproject/rfcs#before-creating-an-rfc) we state that submitters should "lay some groundwork" for their proposals before submitting RFCs in order to prevent them from being rejected. While we have yet to really define what that threshold is, I admit that there's some allure to the idea of testing the waters with a "quick" post rather than a formal RFC (an RFC involves forking the repo, creating a branch, filling out sections of the template, and submitting a PR, while a post on the Discussions page is free text that can be made in the browser).

I guess the TL;DR of this is that if anything, I think it should replace the General Development forums, and if we need help deciding if that's a good idea, we can check the [General Development forums](https://obsproject.com/forum/list/general-development.21/) now to see if that is content we would rather have over here. I think we could split development discussion into Core Development, Plugins, and Scripts, which might be more useful for filtering the content of that forum a bit more. The other challenge might be training GitHub users to use the discussions section instead of submitting Issues when they have build questions, for example. Might also be worthwhile to have a Feature Development tag to host discussion of potential feature development in the pre-RFC stage, if that's useful at all.

**Reply by VodBox** [reply]
I'm of the mind that GitHub Discussions should be a replacement for our General Development forum, absolutely. I'm much more aware of what happens on GitHub than the forums, and the ability to turn threads in issues or PRs into a discussion will help give us an option to move things to a separate discussion thread without requiring people sign up for our forums or join the Discord.

Just the fact that there are people who join our forum who will never join Discord, and people who join our Discord who will never sign up for our forum, means that discussion is necessarily fragmented between groups. The vast majority of those people in both places are likely to have a GitHub, or be willing to get a GitHub account. Contributing to the project already requires a GitHub account anyway.
